classes/web_processor.php

```diff
@@ -98,7 +99,7 @@ public function process(manager $manager, bool $isfinal) {
         $this->profile->set('duration', $current - $manager->get_starttime());
         $this->profile->set('maxstackdepth', $this->sampleset->get_stack_depth());
         $reason = $manager->get_reasons($this->profile);
-        if ($reason !== profile::REASON_NONE) {
+        if ($reason !== profile::REASON_NONE && $USER != null) {
```

Comment:
This seems wrong? Why would we not want to save the record if we don't have a user yet? Is this related to this patch?

Reply:
To test this I was putting a sleep() on one page to lock the session and then trying to load another at the same time. I was getting this error when the second page loaded, as it was trying to save a null user. 

![image](https://user-images.githubusercontent.com/63634595/188551024-05482050-3216-4fe4-9be1-fee738c6cc2e.png)

Reply:
This could be because it is trying to save user info before the session has started, therefore no user information is available.

Reply:
As an explicit test scenario I'd like you to test a page which is holding the session lock open in one page for say 10 seconds, and then we are profiling a second page which is blocked by the first page. If you set the numbers really low so that it is saving partial profiles every 1 seconds, then I would expect the profile for page B to exist without any user data after 1 second and to show all of its time spent in the session waiting for a lock. Then after 10 seconds the lock will be released, the user will be logged in, and the profile metadata will be updated to include the user that we now know who they are